Modules/CalcSetup.lua

```diff
@@ -213,349 +354,339 @@ function calcs.initEnv(build, mode, override)
-			end
-			if item.type ~= "Jewel" and item.type ~= "Flask" then
-				-- Update item counts
-				local key
-				if item.rarity == "UNIQUE" or item.rarity == "RELIC" then
-					key = "UniqueItem"
-				elseif item.rarity == "RARE" then
-					key = "RareItem"
-				elseif item.rarity == "MAGIC" then
-					key = "MagicItem"
-				else
-					key = "NormalItem"
-				end
-				env.modDB.multipliers[key] = (env.modDB.multipliers[key] or 0) + 1
-				if item.corrupted then
-					env.modDB.multipliers.CorruptedItem = (env.modDB.multipliers.CorruptedItem or 0) + 1
-				else
-					env.modDB.multipliers.NonCorruptedItem = (env.modDB.multipliers.NonCorruptedItem or 0) + 1
-				end
-				if item.shaper then
-					env.modDB.multipliers.ShaperItem = (env.modDB.multipliers.ShaperItem or 0) + 1
-					env.modDB.conditions["ShaperItemIn"..slotName] = true
-				else
-					env.modDB.multipliers.NonShaperItem = (env.modDB.multipliers.NonShaperItem or 0) + 1
-				end
-				if item.elder then
-					env.modDB.multipliers.ElderItem = (env.modDB.multipliers.ElderItem or 0) + 1
-					env.modDB.conditions["ElderItemIn"..slotName] = true
 				else
-					env.modDB.multipliers.NonElderItem = (env.modDB.multipliers.NonElderItem or 0) + 1
+					env.itemModDB:ScaleAddList(srcList, scale)
 				end
-				if item.shaper or item.elder then
-					env.modDB.multipliers.ShaperOrElderItem = (env.modDB.multipliers.ShaperOrElderItem or 0) + 1
+				if item.type ~= "Jewel" and item.type ~= "Flask" then
+					-- Update item counts
+					local key
+					if item.rarity == "UNIQUE" or item.rarity == "RELIC" then
+						key = "UniqueItem"
+					elseif item.rarity == "RARE" then
+						key = "RareItem"
+					elseif item.rarity == "MAGIC" then
+						key = "MagicItem"
+					else
+						key = "NormalItem"
+					end
+					env.itemModDB.multipliers[key] = (env.itemModDB.multipliers[key] or 0) + 1
+					if item.corrupted then
+						env.itemModDB.multipliers.CorruptedItem = (env.itemModDB.multipliers.CorruptedItem or 0) + 1
+					else
+						env.itemModDB.multipliers.NonCorruptedItem = (env.itemModDB.multipliers.NonCorruptedItem or 0) + 1
+					end
+					if item.shaper then
+						env.itemModDB.multipliers.ShaperItem = (env.itemModDB.multipliers.ShaperItem or 0) + 1
+						env.itemModDB.conditions["ShaperItemIn"..slotName] = true
+					else
+						env.itemModDB.multipliers.NonShaperItem = (env.itemModDB.multipliers.NonShaperItem or 0) + 1
+					end
+					if item.elder then
+						env.itemModDB.multipliers.ElderItem = (env.itemModDB.multipliers.ElderItem or 0) + 1
+						env.itemModDB.conditions["ElderItemIn"..slotName] = true
+					else
+						env.itemModDB.multipliers.NonElderItem = (env.itemModDB.multipliers.NonElderItem or 0) + 1
+					end
+					if item.shaper or item.elder then
+						env.itemModDB.multipliers.ShaperOrElderItem = (env.itemModDB.multipliers.ShaperOrElderItem or 0) + 1
+					end
 				end
 			end
 		end
 	end
 
+	-- Merge env.itemModDB with env.ModDB
+	mergeDB(env.modDB, env.itemModDB)
```

Comment:
It's there to enable re-use when accelerating certain environment modifiers. For example, if I'm looking at impact of a new support gem to put into my skill-group, I don't need to re-calculate the modifiers provided by any items I have since they haven't changed, but I do need a new environment so I can calculated A versus B as the new support gem could impact some of the existing modifiers. This setup allows us to not re-calculated or re-asses modifiers that don't need to be.